holidays/countries/spain.py

```
@@ -140,158 +144,170 @@ def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
-            # Christmas Day.
-            self._add_christmas_day(tr("Natividad del Señor"))
+            self._add_immaculate_conception_day(tr("Inmaculada Concepción")),
+            rule=SUN_TO_NONE,
+        )
 
-    def _populate_subdiv_an_public_holidays(self):
-        if self._year in {2012, 2017, 2023}:
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day(tr("Natividad del Señor")), rule=SUN_TO_NONE)
+
+    def _populate_subdiv_holidays(self):
```

Ruff (0.14.2)

Missing return type annotation for private function `_populate_subdiv_holidays`

Add return type annotation: `None`

(ANN202)